DTDL/tcads_stream.json

```diff
@@ -0,0 +1,25 @@
+{
+    "@context" : "dtmi:dtdl:context;2",
+    "@id" : "dtmi:io:tributech:stream:tcads;1", 
+    "@type": "Interface", 
+    "extends": "dtmi:io:tributech:stream:base;1",
+    "displayName": "Beckhoff TwinCat ADS Stream", 
+    "contents" : [
+        {
+            "@type" : "Property", 
+            "name" : "Identifier", 
+            "schema" : "string", 
+            "writeable" : true, 
+            "displayName" : "Symbol Path", 
+            "description" : "The Symbol Path for the Twin Cat ADS Source (eg. \"Main.iTest\")"
+        },
+        {
+            "@type" : "Property", 
+            "name" : "Datatype", 
+            "schema" : "string", 
```

Comment:
Should we maybe use an [enum](https://github.com/Azure/opendigitaltwins-dtdl/blob/master/DTDL/v2/dtdlv2.md#enum) and restrict it to the currently supported types?
Enums might make it easier to build (internal) conversion logic.

Reply:
Thought about that but since you can choose a Random Name for the Datatype and still have a valid Datatype i am not quite sure about a predefined List of Datatypes. You can set up a DataType INT123 and have it a value of int16. I mean it should not happen but you never know ? 

Whats your opinion on that ?

Reply:
Aren't we referencing to this predefined data types?
https://infosys.beckhoff.com/content/1033/tc3_system/html/tcsysmgr_datatypecomparison.htm?id=3043404538898382042

...which are assigned to variables
https://infosys.beckhoff.com/content/1033/tc3_system/html/tcsysmgr_varinfo.htm?id=5356958998684184874


Or is it about such "custom" types (which seem to reference specific types under the hood)?
https://infosys.beckhoff.com/content/1033/typesystem/9007202793718155.html?id=7503826230936521275

Reply:
In the underlying System there will be always the predefined DataTypes. But you can rename those DataTypes and use them for other Variables. For Example you can use an DWORD  for a "Time" DataType or you can use a DINT for the same Type or you can rename DWORD and name it "EXAMPLETYPE" and under the hood it will always use the UINT32. So you can define your own Types which will always have the predefinied Types under the Hood. Thats why i used only a string for the flexability. It's all about how you delegate the real Datatype to the Source, because we can read the underlying Datatype and use those Information for parsing our Values. Or we decide that we only offer the Standard Datatypes and if we have to support Custom Types we have to implement a new Model or add it into the Enum. But we dont need to read the Structure of the Variable for that, because we know the Types.

You can see it here : 
https://infosys.beckhoff.com/english.php?content=../content/1033/tc3_system/html/TcSysMgr_DatatypeComparison.htm&id=

Reply:
If our source will (easily) be able to retrieve the actual, pre-defined type of a custom named type from the Beckhoff system and convert it to one of our [supported encoding types](https://github.com/tributech-solutions/data-asset-twin/blob/ef03590838d19caaabed76bd9e24967749800444/DTDL/base_stream.json#L67) it's fine for me!?

Reply:
Lets use the Enum Property if we want to support Costum Types, we will have to reconsider this again. I think , on the long run, it will be the safer choice.